### PR TITLE
TASK: Bring HTTP docs up to date with actual implementation

### PR DESCRIPTION
**Pull request by albe**
Start of fixing/improving the HTTP docs to match actual implementation as of Flow 6.0 / PSR-7

### CODE REVIEW

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 1]
```diff
@@ -31,7 +31,7 @@ The basic walk through a Flow-based web application is as follows:
 * by default, the :abbr:`HTTP Request Handler (\\Neos\\Flow\\Http\\RequestHandler)` takes over and runs a boot sequence
   which initializes all important parts of Flow
 * the HTTP Request Handler builds an PSR-7 HTTP Request and Response object. The
-  :abbr:`Request object (\\Neos\\Flow\\Http\\Request)` contains all important properties of the real HTTP request.
+  :abbr:`Request object (\\Psr\\Http\\Request)` contains all important properties of the real HTTP request.
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:34]:
```suggestion
  :abbr:`Request object (\\Psr\\Http\\Message\\ServerRequestInterface)` contains all important properties of the real HTTP request.
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 2]
```diff
@@ -31,7 +31,7 @@ The basic walk through a Flow-based web application is as follows:
...
   The :abbr:`Response object (\\Neos\\Flow\\Http\\Response)` in turn is empty and will be filled with information by a
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:35]:
```suggestion
  The :abbr:`Response object (\\Psr\Http\\Message\\ResponseInterface)` in turn is empty and will be filled with information by a
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 3]
```diff
@@ -205,52 +205,25 @@ still handled even if the new component cancels the current chain.
...
-The second method, ``createFromEnvironment()``, take the environment provided by PHP's superglobals and specialized
-functions into account. It creates a ``Request`` instance which reflects the current HTTP request received from the web
-server. This method is best used if you need a ``Request`` object with all properties set according to the current
-server environment and incoming HTTP request.
-Note though, that you should not expect this ``Request`` to match the current ``Request``, since the latter will still
-have been affected by some ``HTTP Components``. If you need the **current** Request, get it from the ``RequestHandler`` instead.
+In the PSR-7 specification, a distinction is made between two different types of requests - incoming (``ServerRequest``)
+and outgoing (``Request``). Whenever you want to make an outgoing request, you can easily use for example Guzzle's
+``Request`` class constructor with the respective arguments for method, uri, etc. and then pass that to e.g. a PSR-18
+Http Client implementation.
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:211]: We need to make our own `Browser` PSR-18 compatible at some point. Right now it still uses a `ServerRequest` though in order to provide easy request forwarding in functional tests.

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 4]
```diff
@@ -205,52 +205,25 @@ still handled even if the new component cancels the current chain.
...
+On the other side the incoming request is something you should never try to create an instance of yourself, as it is
+provided by the framework. In theory, you could also call inject the ``ServerRequestFactory::createServerRequest`` or
+Guzzle's ``ServerRequest::fromGlobals()`` convenience method, but this does not have any relation to the current request
+object handled by the framework. It will not have any of the processing from components applied and might therefore lead
+to unexpected results, like the trusted proxy headers ``X-Forwarded-*`` not being applied and the ``ServerRequest`` providing
+wrong protocol, host or client IP address.
+If you need access to the **current** HTTP ``Request``, either create a Http Component or only access it inside the
```
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:218]: This is in preparation of the deprecation of `ComponentContext` with PSR-15

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 5]
```diff
@@ -205,52 +205,25 @@ still handled even if the new component cancels the current chain.
...
-
-The second method, ``createFromEnvironment()``, take the environment provided by PHP's superglobals and specialized
-functions into account. It creates a ``Request`` instance which reflects the current HTTP request received from the web
-server. This method is best used if you need a ``Request`` object with all properties set according to the current
-server environment and incoming HTTP request.
-Note though, that you should not expect this ``Request`` to match the current ``Request``, since the latter will still
-have been affected by some ``HTTP Components``. If you need the **current** Request, get it from the ``RequestHandler`` instead.
+In the PSR-7 specification, a distinction is made between two different types of requests - incoming (``ServerRequest``)
+and outgoing (``Request``). Whenever you want to make an outgoing request, you can easily use for example Guzzle's
+``Request`` class constructor with the respective arguments for method, uri, etc. and then pass that to e.g. a PSR-18
```
Comment (bwaidelich) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:210]: tiny wording improvement
```suggestion
and outgoing (``Request``). Whenever you want to make an outgoing request, you can easily use the Guzzle
``Request`` class constructor for example with the respective arguments for method, uri, etc. and then pass that to e.g. a PSR-18
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 6]
```diff
@@ -205,52 +205,25 @@ still handled even if the new component cancels the current chain.
...
+Http Client implementation.
+On the other side the incoming request is something you should never try to create an instance of yourself, as it is
+provided by the framework. In theory, you could also call inject the ``ServerRequestFactory::createServerRequest`` or
```
Comment (bwaidelich) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:213]:
```suggestion
provided by the framework. In theory, you could also call the ``ServerRequestFactory::createServerRequest`` or
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 7]
```diff
@@ -205,52 +205,25 @@ still handled even if the new component cancels the current chain.
...
+Guzzle's ``ServerRequest::fromGlobals()`` convenience method, but this does not have any relation to the current request
```
Comment (bwaidelich) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:214]:
```suggestion
the Guzzle ``ServerRequest::fromGlobals()`` convenience method, but this does not have any relation to the current request
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 8]
```diff
@@ -205,52 +205,25 @@ still handled even if the new component cancels the current chain.
...
+object handled by the framework. It will not have any of the processing from components applied and might therefore lead
+to unexpected results, like the trusted proxy headers ``X-Forwarded-*`` not being applied and the ``ServerRequest`` providing
+wrong protocol, host or client IP address.
+If you need access to the **current** HTTP ``Request``, either create a Http Component or only access it inside the
+controller through the ``ActionRequest`` for inspecting.
```
Comment (bwaidelich) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:219]: A link to the HTTP Component section and a mini example for the ActionRequest case would be rad

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst [hunk 9]
```diff
@@ -268,27 +241,29 @@ Such a request is always bound to an ``Http\Request``::
...
 
-The request features a few methods for retrieving and setting arguments. These arguments are the result of merging any
-GET, POST and PUT arguments and even the information about uploaded files. Be aware that these arguments have not been
-sanitized or further processed and thus are not suitable for being used in controller actions. If you, however, need to
-access the raw data, these API function are the right way to retrieve them.
+The ``ActionRequest`` features a few methods for retrieving and setting arguments. These arguments are the result of merging any
+GET, POST and PUT arguments and even the information about uploaded files. Note that these arguments have already been processed
+by the validation and property mapping layerns and thus are suitable for being used in controller actions. If you, however, need to
+access the raw data, you can access these via the ``HttpRequest`` ``getCookieParams()``, ``getQueryParams()``, ``getUploadedFiles()``
+and ``getParsedBody()`` methods respectively.
```
Comment (bwaidelich) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:255]:
```suggestion
access the raw data, you can access these via the ``getCookieParams()``, ``getQueryParams()``, ``getUploadedFiles()`` and ``getParsedBody()``
methods of the ``HttpRequest``  respectively.
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst (hunk 4 above)
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:218]:
```suggestion
If you need access to the **current** HTTP ``Request``, either create a :ref:`Http Component<Component Chain>` or only access it inside the
```

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst (hunk 8 above)
Comment (albe) [inline Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:219]:
```suggestion
controller through the ``ActionRequest`` for inspecting::

	public function myAction() {
		$requestBody = $this->request->getHttpRequest()->getParsedBody();
		...
	}
```
Not sure that's what you were aiming for?

**Comment by albe** [comment]
btw - who can update the flow-graph images? :)

File: Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst (hunk 8 above)
Comment (bwaidelich) [reply Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Http.rst:219]: Yes, great, thanks :)

**Comment by bwaidelich** [comment]
>  who can update the flow-graph images?

There should be corresponding xml files in the same directory that can be loaded in draw.io but I can check tomorrow!

**Comment by bwaidelich** [comment]
@albe I forgot to check, but the draw.io files are indeed inside the Images folder: https://github.com/neos/flow-development-collection/tree/master/Neos.Flow/Documentation/TheDefinitiveGuide/PartIII/Images

**Comment by albe** [comment]
I'll merge this for now, as I want to have a bit more accurate docs with the 6.2 release at least. Let's fix the graphs in a follow-up sometime

**Pull request merged**